cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -167,9 +167,27 @@ message VolumeRequest {
         int64 seismic_store_id = 2;  // Need to be a data manager or tenant user, not 3rd party, to access by tracestore
     }
     LineBasedVolume volume = 3;
+    GeometryFilter geometry_filter = 5; // Mutually exclusive with volume
     bool include_trace_header = 4;
 }
 
+message GeometryFilter {
+    oneof filter {
+        LineSlice slice = 1;
+        VolumeCube cube = 2;
+    }
+    LineDescriptor z_range = 3;
+}
+
+message LineSlice {
+    Geometry line = 1;
+    InterpolationMethod interpolation_method = 2;
+}
+
+message VolumeCube {
```

Comment:
Is volume cube a bit misleading for this type? We tend to use the word cube quite a lot for volume requests so a name as generic as this might get a bit confusing. Maybe these type should be named something like `GeometryCubeFilter` and `GeometryLineFilter`?